iib/workers/tasks/utils.py

```diff
@@ -526,7 +528,17 @@ def set_registry_token(token, container_image):
 
     registry = ImageName.parse(container_image).registry
     encoded_token = base64.b64encode(token.encode('utf-8')).decode('utf-8')
-    registry_auths = {'auths': {registry: {'auth': encoded_token}}}
+    registry_auths = {'auths': {}}
+    if append:
+        docker_config_path = os.path.join(os.path.expanduser('~'), '.docker', 'config.json')
+        if os.path.exists(docker_config_path):
+            with open(docker_config_path, 'r') as f:
+                registry_auths = json.load(f)
```

Comment:
I am not sure how config.json file is created, however I would add here TypeError catch in case that json.loads() would fail to read the file.

Reply:
I've added catching exception an logging it. However I raised it again because I think we want to fail horribly when the config is incorrect.